pkg/authentication/externalidp/awsiam/aws_client.go

```diff
@@ -0,0 +1,153 @@
+		return nil, fmt.Errorf("generating sts presigned url: %w", err)
+	}
+	tokenInfo, err := ParsePresignedURL(url)
+	if err != nil {
+		return nil, fmt.Errorf("parsing credentials sts presigned url: %w", err)
+	}
+	return tokenInfo, nil
+}
+
+func ParsePresignedURL(presignedURL string) (*AWSIdentityTokenInfo, error) {
+	parsedURL, err := url.Parse(presignedURL)
+	if err != nil {
+		return nil, err
+	}
+
+	queryParams := parsedURL.Query()
+	credentials := queryParams.Get(AuthCredentialKey)
+	splitedCreds := strings.Split(credentials, "/")
+	if len(splitedCreds) < mimLengthSplitCreds {
+		return nil, fmt.Errorf("invalid credentials format: %w", ErrInvalidCredentialsFormat)
+	}
+	calculatedRegion := splitedCreds[2]
+	return &AWSIdentityTokenInfo{
+		Method:             "POST",
+		Host:               parsedURL.Host,
+		Region:             calculatedRegion,
+		Action:             AuthAction,
+		Date:               queryParams.Get(AuthDateKey),
+		ExpirationDuration: queryParams.Get(AuthExpiresKey),
+		AccessKeyID:        splitedCreds[0],
+		Signature:          queryParams.Get(AuthSignatureKey),
+		SignedHeaders:      strings.Split(queryParams.Get(AuthSignedHeadersKey), ";"),
+		Version:            queryParams.Get(AuthVersionKey),
+		Algorithm:          queryParams.Get(AuthAlgorithmKey),
+		SecurityToken:      queryParams.Get(AuthSecurityTokenKey),
+	}, nil
+}
+
+func PresignGetCallerIdentityFromAuthParams(ctx context.Context, params *IAMAuthParams, stsClient *sts.Client) (string, error) {
+	setHTTPHeaders := func(requestHeaders map[string]string, ttl time.Duration) func(*middleware.Stack) error {
+		middlewareName := "AddHeaders" + uuid.New().String()
+		return func(stack *middleware.Stack) error {
+			return stack.Build.Add(middleware.BuildMiddlewareFunc(middlewareName, func(
+				ctx context.Context, in middleware.BuildInput, next middleware.BuildHandler,
+			) (
+				middleware.BuildOutput, middleware.Metadata, error,
+			) {
+				if req, ok := in.Request.(*smithyhttp.Request); ok {
+					req.Method = "POST"
+					for header, value := range requestHeaders {
+						req.Header.Add(header, value)
+					}
+					queryParams := req.URL.Query()
+					queryParams.Set(AuthExpiresKey, fmt.Sprintf("%d", int(ttl.Seconds())))
+					req.URL.RawQuery = queryParams.Encode()
+				}
+				return next.HandleBuild(ctx, in)
+			}), middleware.Before)
+		}
+	}
+
+	stsPresignClient := sts.NewPresignClient(stsClient, func(o *sts.PresignOptions) {
+		o.ClientOptions = append(o.ClientOptions, func(opts *sts.Options) {
+			opts.ClientLogMode = aws.LogSigning
+		})
+	})
+
+	presign, err := stsPresignClient.PresignGetCallerIdentity(context.Background(), &sts.GetCallerIdentityInput{},
+		sts.WithPresignClientFromClientOptions(sts.WithAPIOptions(setHTTPHeaders(params.TokenRequestHeaders, params.URLPresignTTL))),
+	)
+	if err != nil {
+		return "", err
+	}
+	return presign.URL, err
+}
+
+func RetrieveCredentials(ctx context.Context, cfg *aws.Config) (*aws.Credentials, error) {
```

Comment:
I'd strongly consider removing this function. 
After our ping-pong's it's been reduced to really 1 line of code `creds, err := cfg.Credentials.Retrieve(ctx)` , not used in lakeFS. 
It has no extra logic so the tests are essentially just testing AWS sdk `cfg.Credentials.Retrieve` function that we don't need to test. 

In other words, a whole lot of bloat to the code for a single func call that is not implemented by us anyway.